compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeUIViewController.uikit.kt

```diff
@@ -0,0 +1,406 @@
+            it.viewWillLayoutSubviews()
+        }
+    }
+
+    override fun viewWillTransitionToSize(
+        size: CValue<CGSize>,
+        withTransitionCoordinator: UIViewControllerTransitionCoordinatorProtocol
+    ) {
+        super.viewWillTransitionToSize(size, withTransitionCoordinator)
+
+        if (isInsideSwiftUI || presentingViewController != null) {
+            // SwiftUI will do full layout and scene constraints update on each frame of orientation change animation
+            // This logic is not needed
+
+            // When presented modally, UIKit performs non-trivial hierarchy update durting orientation change,
+            // its logic is not feasible to integrate into
+            return
+        }
+
+        // Happens during orientation change from LandscapeLeft to LandscapeRight, for example
+        val isSameSizeTransition = view.frame.useContents {
+            CGSizeEqualToSize(size, this.size.readValue())
+        }
+        if (isSameSizeTransition) {
+            return
+        }
+
+        composeSceneMediator?.viewWillTransitionToSize(
+            targetSize = size,
+            coordinator = withTransitionCoordinator,
+        )
+        layers.fastForEach {
+            it.viewWillTransitionToSize(
+                targetSize = size,
+                coordinator = withTransitionCoordinator,
+            )
+        }
+        view.layoutIfNeeded()
+    }
+
+    override fun viewWillAppear(animated: Boolean) {
+        super.viewWillAppear(animated)
+
+        isInsideSwiftUI = checkIfInsideSwiftUI()
+        setContent(content)
+        configuration.delegate.viewWillAppear(animated)
+    }
+
+    override fun viewDidAppear(animated: Boolean) {
+        super.viewDidAppear(animated)
+        composeSceneMediator?.viewDidAppear(animated)
+        layers.fastForEach {
+            it.viewDidAppear(animated)
+        }
+        configuration.delegate.viewDidAppear(animated)
+    }
+
+    // viewDidUnload() is deprecated and not called.
+    override fun viewWillDisappear(animated: Boolean) {
+        super.viewWillDisappear(animated)
+        composeSceneMediator?.viewWillDisappear(animated)
+        layers.fastForEach {
+            it.viewWillDisappear(animated)
+        }
+        configuration.delegate.viewWillDisappear(animated)
+    }
+
+    override fun viewDidDisappear(animated: Boolean) {
+        super.viewDidDisappear(animated)
+
+        dispose()
+
+        dispatch_async(dispatch_get_main_queue()) {
+            kotlin.native.internal.GC.collect()
+        }
+
+        configuration.delegate.viewDidDisappear(animated)
+    }
+
+    override fun didReceiveMemoryWarning() {
+        println("didReceiveMemoryWarning")
+        kotlin.native.internal.GC.collect()
+        super.didReceiveMemoryWarning()
+    }
+
+    @OptIn(ExperimentalComposeApi::class)
+    fun setContent(content: @Composable () -> Unit) {
+        if (composeSceneMediator != null) {
+            return // already attached
```

Comment:
You still need to update content in the existing mediator

Reply:
done